### ISSUE TITLE
Improve parse_duration error when missing units

### ISSUE THREAD

**Issue by dtolnay** [tailhook/humantime#6]
```rust
fn main() {
    println!("{}", humantime::parse_duration("100").unwrap_err());
}
```

```console
$ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/testing`
unknown unit at 3-3
```

The error message says "unknown unit at 3-3" which is not particularly understandable if someone isn't already familiar with humantime. It would be better to have the message recognize the specific common case of digits-only and produce a targeted message that includes the caller's number with some illustrative unit suffixes. Maybe: "needs time units, for example 100ms or 100sec" where the number is always the same one the caller passed.

A better error message would be a big improvement when using humantime in a cli, such as with structopt:

```rust
#[derive(StructOpt)]
struct Args {
    ...

    /// Timeout for operation O, e.g. 30sec or 2min
    #[structopt(
        long,
        default_value = "5min",
        parse(try_from_str = "humantime::parse_duration")
    )]
    timeout: Duration,
}
```

```console
$ ./cli --timeout 100
error: Invalid value for '--timeout <timeout>': unknown unit at 3-3
```

**Comment by tailhook** [comment]
@dtolnay please, take a look at #6. Is it good enough?

**Comment by dtolnay** [comment]
I would like for it to be better.

- "time unit needed at 3" isn't terminology that would make sense to a user of a cli. Please leave off "at 3" because it serves only to confuse. It's confusing if the user didn't write 3 anywhere, and it's confusing if the user *did* write a 3 in their input, so it is just confusing in general.

- Please consider using the actual number that the caller provided when showing the example suffixes, instead of the arbitrary number 100.

- "unknown time unit at 3-8" is similarly confusing, please show the actual time unit that they typed that was not recognized.

**Comment by tailhook** [comment]
Wanted to keep backward compatibility and keep error structure smaller, but that probably doesn't matter.

Please take another look: https://github.com/tailhook/humantime/commit/49f11fdc2a59746085d2457cb46bce204dec746a

**Comment by dtolnay** [comment]
Thanks, that looks good to me!

**Comment by tailhook** [comment]
Released in v2.0.0
